Source/WebCore/rendering/RenderTable.cpp

```diff
@@ -1474,7 +1474,11 @@ RenderTableSection* RenderTable::bottomSection() const
     recalcSectionsIfNeeded();
     if (m_foot)
         return m_foot.get();
+    if (m_head && !firstBody())
+        return m_head.get();
     for (RenderObject* child = lastChild(); child; child = child->previousSibling()) {
+        if (child == m_head)
+            continue;
         if (auto* tableSection = dynamicDowncast<RenderTableSection>(*child))
             return tableSection;
     }
```

Comment:
This looks good, though I would move the entire function next to `RenderTable::topSection` and mirror its logic something along the lines of.

```
RenderTableSection* RenderTable::bottomSection() const
{
    recalcSectionsIfNeeded();
    if (m_foot)
        return m_foot.get();
    for (CheckedPtr child = lastChild(); child; child = child->previousSibling()) {
        if (child.get() == m_head.get())
            continue;
        if (auto* tableSection = dynamicDowncast<RenderTableSection>(*child))
            return tableSection;
    }
    return m_head.get();
}

```

so it's like, return footer first, body second and head last.

Reply:
moved and changed!
but github issues for now to send the PR.